### PR TITLE
Add camel-ai dataset

### LINKED ISSUE #2501: Dataset: CAMEL Math and Physics
50K+ Math, Physics and Code inputs and outputs, sounds tasty but hidden in a zip file.

https://github.com/lightaime/camel#data-hosted-on-hugging-face

### PR DESCRIPTION
**Pull request by CheckMC**
Added the oa_camel folder in data. 

This dataset is quite large, at 110,000 entries. The original source is found on their page on HF: https://huggingface.co/camel-ai

It contains question-answer pairs about physics, chemisty, biology, and math, all combined into one dataset in the OA format.

Closes #2501. The dataset is uploaded to my HF at: https://huggingface.co/datasets/checkai/oaCamel

### CODE REVIEW

File: data/datasets/oa_camel/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Languages
+English
+
+Dataset Structure
+This dataset follows the OA format, which is:
+
+INSTRUCTION (string): Instruction text
+RESPONSE (string): Expected response to the instruction
+SOURCE (string): Original data source short name, e.g. "wikipedia"
```
Comment (sedthh) [inline data/datasets/oa_camel/README.md:14]: Please describe the contents of the columns for CamelAI instead (like the keys and values for METADATA)

File: data/datasets/oa_camel/prepare.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# This program prepares Camel datasets to the Open Assistant format.
+
+import pandas as pd
+from datasets import load_dataset
+import json
+
+# Load the datasets from Hugging Face
+datasets = [
+  'camel-ai/biology', 'camel-ai/physics', 'camel-ai/chemistry', 'camel-ai/math'
```
Comment (sedthh) [inline data/datasets/oa_camel/prepare.py:9]: Can you please explain if CamelAI has given written access to using their repository? Is this dataset for research purposes only or can be used for all purposes?

File: data/datasets/oa_camel/prepare.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      'SOURCE': dataset_id,
+      'METADATA': json.dumps(metadata)
+    }
+    transformed_data.append(transformed_entry)
+
+# Combine the results into a single DataFrame
+df = pd.DataFrame(transformed_data)
+
+# Save the DataFrame to disk in the Parquet format
+df.to_parquet('output.parquet', row_group_size=100)
```
Comment (sedthh) [inline data/datasets/oa_camel/prepare.py:31]: please save and reupload the dataset in the following format:
`df.to_parquet("output.parquet", row_group_size=100, engine="pyarrow", index=False)`

**Review by sedthh — CHANGES_REQUESTED** [review]
Really useful instruction dataset. Please also run `pre-commit run` after fixing the requested changes. Thanks!

**Comment by CheckMC** [comment]
due to license issue and the dataset being 'research' only, I'm scrapping this dataset

**Pull request closed without merging**